src/Security/Validation/AbstractPasswordValidator.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace SilverStripe\Security\Validation;
+
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Core\Validation\ValidationResult;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\MemberPassword;
+
+/**
+ * Abstract validator with functionality for checking for reusing old passwords.
+ */
+abstract class PasswordValidator
+{
+    use Injectable;
+    use Configurable;
+
+    /**
+     * Default number of previous passwords to check for a reusing old passwords.
+     */
+    private static int $historic_count = 6;
+
+    protected ?int $historicalPasswordCount = null;
+
+    public function validate(string $password, Member $member): ValidationResult
+    {
+        $result = ValidationResult::create();
+
+        $historicCount = $this->getHistoricCount();
+        if ($historicCount) {
+            $idColumn = DataObject::getSchema()->sqlColumnForField(MemberPassword::class, 'MemberID');
+            $previousPasswords = MemberPassword::get()
+                ->where([$idColumn => $member->ID])
+                ->sort(['Created' => 'DESC', 'ID' => 'DESC'])
```

Comment:
```suggestion
                ->sort(['ID' => 'DESC'])
```
Not sure why we have 'Created' in there

Reply:
That was there from code copied from the original password validator. It's there so we're checking the last x passwords, not just some arbitrary previous passwords.